editor/plugins/node_3d_editor_plugin.cpp

```diff
@@ -1459,14 +1459,15 @@ Transform3D Node3DEditorViewport::_compute_transform(TransformMode p_mode, const
 				p_motion.snap(Vector3(p_extra, p_extra, p_extra));
 			}
 
+			Transform3D t;
 			if (p_local) {
-				p_motion = p_original.basis.xform(p_motion);
+				t = p_original_local;
+				t.origin += p_original_local.basis.xform(p_motion);
```

Comment:
This matches the other cases and makes the movement smooth, using `p_original` makes the movement jerky with scale, which doesn't make sense to me, for example if it is scaled the motion is scaled up in the original implementation